### PR TITLE
Improve HTTP error handling and update documentation

### PR DESCRIPTION
**Pull request by lcbm**
This pr introduces three changes:

- Once again, `createUser()` returns the user credentials to the caller;
- Improved error handling for HTTP operations;
- Update documentation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,202 +1,147 @@
...
 
 ```javascript
-const KNoTCloudAuthenticator = require('@cesarbr/knot-cloud-sdk-js-authenticator');
+const KNoTCloudAuthenticator = require("@cesarbr/knot-cloud-sdk-js-authenticator");
 
 const client = new KNoTCloudAuthenticator({
-  protocol: 'https',
-  hostname: 'auth.knot.cloud',
+  protocol: "https",
+  hostname: "auth.knot.cloud",
```
Comment (netoax) [inline README.md]: Please, change to `api.knot.cloud` considering these operations are routed by the API Gateway.

File: README.md [hunk 2]
```diff
@@ -1,202 +1,147 @@
...
+  port: 8081,
```
Comment (netoax) [inline README.md]: I think you can omit this port considering the 443 port is inferred from the `protocol` property. Another option is to put 443 explicitly.

File: README.md [hunk 3]
```diff
@@ -1,202 +1,147 @@
...
-  * `hostname` **String** KNoT Cloud authenticator instance host name.
-  * `port` **Number** (Optional) KNoT Cloud authenticator instance port. Default: 443.
-  * `pathname` **String** (Optional) Path name on the server.
+#### Arguments
 
-### Example
+- `config` **Object** JSON object with request details.
+  - `protocol` **String** (Optional) Either `'http'` or `'https'`. Default: `'https'`.
+  - `hostname` **String** KNoT Cloud authenticator instance host name.
+  - `port` **Number** (Optional) KNoT Cloud authenticator instance port. Default: 8081.
```
Comment (netoax) [inline README.md]: Default to 80/443.

**Review by netoax — CHANGES_REQUESTED** [review]
LGTM at all, thanks!
Just some comments about the README changes.

File: README.md (hunk 2 above)
Comment (lcbm) [reply README.md:23]: Shouldn't it be `80`?

File: README.md (hunk 1 above)
Comment (lcbm) [reply README.md]: Done!

File: README.md (hunk 2 above)
Comment (lcbm) [reply README.md]: Done!

File: README.md (hunk 3 above)
Comment (lcbm) [reply README.md]: Done!

**Review by netoax — APPROVED** [review]

**Pull request merged**